### PR TITLE
Adjust the test suite to have a separate pytest style tests directory

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

**Review by cmcmarrow — DISMISSED** [review]

File: tests/nts/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# New Test Suite
+
+Welcome to the *new* test suite for Salt!
+
+Any test under this directory shall be written exploring the full capabilities of PyTest.
+That means, no occurrences of the [TestCase](https://docs.python.org/3/library/unittest.html#unittest.TestCase) class
+shall be used, neither our [customizations to it](../support/case.py).
```
Comment (waynew) [inline tests/nts/README.md:7]: Would it be possible/simple to enforce this via a test? I'd be 👍 for that.

Comment (s0undt3ch) [reply tests/nts/README.md:7]: You mean force new tests to be on the PyTest only directory?
Once we fully switch we could make a pre-commit hook to enforce that.

Comment (waynew) [reply tests/nts/README.md:7]: no - you say that `TestCase` and `support.case` shouldn't be used in the new directory. Can we easily enforce *that*?

**Comment by s0undt3ch** [comment]
@dwoz I assume the label was automation?
Because yes, there's a failure under PyTest and it doesn't need to be fixed on this PR.

File: tests/nts/README.md (hunk 1 above)
Comment (s0undt3ch) [reply tests/nts/README.md:7]: Ah! Yes we can!

Comment (s0undt3ch) [reply tests/nts/README.md:7]: Done!

**Review by dwoz — APPROVED** [review]

**Pull request merged**